lib/turn.rb

```diff
@@ -0,0 +1,35 @@
+class Turn
+  GAME_RULES = {
+    rock: { rock: :draw, paper: :lose, scissors: :win },
+    paper: { rock: :win, paper: :draw, scissors: :lose },
+    scissors: { rock: :lose, paper: :win, scissors: :draw }
+  }
+
+attr_reader :player_name, :player_move, :computer_move
+
+  def initialize(options)
+    @player_name = options["player_name"]
+    @player_move = options["player_move"]
+    @computer_move = options["computer_move"]
```

Comment:
When you define Turn.new in the app.rb how are the above variables assigned? Because you have put Turn.new(session), whenever you assign something to session[:whatever] does this automatically enter your @turn?

Reply:
It does indeed! Because the initialize argument is options. The options is then assigned the [player_name] etc. and so this then relates to the session when you put the session as the argument. So then each parameter of the session is then assigned to the correct variables. Hope this makes a little bit of sense? Haha!